Source/Modules/php.cxx

```diff
@@ -133,6 +140,33 @@ extern "C" {
   static void (*r_prevtracefunc) (const SwigType *t, String *mangled, String *clientdata) = 0;
 }
 
+/* Function used to get the return type mapped to PHP return type.
+ * Used for defining class constants or class properties.
+ */
+static String *Swig_Get_type(String *type) {
+  String *return_value = NULL;
+  if ( Cmp(type,"int") == 0 ||
+        Cmp(type,"unsigned int") == 0 ||
+         Cmp(type,"signed int") == 0 ||
+          Cmp(type,"short") == 0 ||
+           Cmp(type,"unsigned short int") == 0 ||
+            Cmp(type,"signed short int") == 0 ||
+             Cmp(type,"long int") == 0 ||
+              Cmp(type,"unsigned long int") == 0 ||
+               Cmp(type,"signed long int") == 0)
+     return_value = NewString("long");
+    else if (Cmp(type,"float") == 0 ||
+              Cmp(type,"double") == 0 ||
+               Cmp(type,"long double") == 0)
+      return_value = NewString("double");
+    else if (Cmp(type,"bool") == 0)
+      return_value = NewString("bool");
+    else if (Cmp(type,"char") == 0 ||
+              Cmp(type,"String") == 0)
+      return_value = NewString("string");
+  return return_value;
+}
```

Comment:
Hey Olly. I'm not exactly handling the constants here. This is used to declare properties in minit section. I just wanted the conversion of standard datatypes available in C++ to respective datatypes supported in zval. This is used in zend_declare properties. All integers types are covered by long, decimal types by double, and bool, string in the zval. So I had to hard-code this rundown conversion.